Automattic/sensei
# Update students report to work in environments that don't support users table relationship

Fixes #5330

### Changes proposed in this Pull Request

* In order to support environments where the users table is separate from the others, it creates a kind of fallback approach, where (only for these cases) it removes the last activity filter and sorting from the reports.
* Also, now it gets the students based on the enrollment calculation user meta.

### Testing instructions


It's not easy to simulate the original issue without using an environment with this specificity. But we can test some other things:

* Enroll some users to some courses to make sure you have a report.
* Add this filter to your env to simulate an environment that can't query relating users table with others: `add_filter( 'sensei_can_use_users_relationship', '__return_false' );`
* Open the reports.
* Make sure you don't see the fields to filter the last activity date, and you can't sort by Lact Activity column.


### New/Updated Hooks

* `sensei_can_use_users_relationship` - Filters if site environment is able make queries including a relationship between users table and others.
* `sensei_students_report_last_activity_filter_enabled` - Filters if the last activity filter by date and the last activity sorting are enabled in the students report.


### Deprecated Code

* `Sensei_Reports_Overview_Data_Provider_Students::only_course_enrolled_users`


### Screenshot / Video

<img width="1011" alt="Screen Shot 2022-09-01 at 17 03 08" src="https://user-images.githubusercontent.com/876340/188002045-d4d56456-68fe-4665-b8d9-8ce59dc88536.png">
